test/integration/reindex-blocked.js

```diff
@@ -0,0 +1,181 @@
+    keys: aliceKey,
+    timeout: CONNECTION_TIMEOUT,
+    replicationScheduler: {
+      autostart: true,
+      debouncePeriod: 0,
+    },
+    ebt: {
+      logging: true
+    }
+  })
+
+  const bobKey = ssbKeys.generate(null, 'bob')
+  const bob = createSsbServer({
+    path: path.join(os.tmpdir(), 'server-bob'),
+    keys: bobKey,
+    timeout: CONNECTION_TIMEOUT,
+    replicationScheduler: {
+      autostart: true,
+      debouncePeriod: 0,
+    },
+    ebt: {
+      logging: true
+    }
+  })
+
+  const doloresKey = ssbKeys.generate(null, 'dolores')
+  const dolores = createSsbServer({
+    path: path.join(os.tmpdir(), 'server-dolores'),
+    keys: doloresKey,
+    timeout: CONNECTION_TIMEOUT,
+    replicationScheduler: {
+      autostart: true,
+      debouncePeriod: 0,
+    },
+    ebt: {
+      logging: true
+    }
+  })
+  
+  await sleep(500)
+
+  t.pass('alice:' + alice.id)
+  t.pass('bob:' + bob.id)
+  t.pass('dolores:' + dolores.id)
+
+  await Promise.all([
+    pify(alice.db.publish)({ type: 'post', text: 'hello A' }),
+    pify(bob.db.publish)({ type: 'post', text: 'hello B' }),
+    pify(dolores.db.publish)({ type: 'post', text: 'hello D' }),
+
+    pify(alice.db.publish)({
+      type: 'contact',
+      contact: bob.Id,
+      following: true,
+    }),
+    pify(bob.db.publish)({
+      type: 'contact',
+      contact: alice.id,
+      following: true,
+    }),
+    pify(dolores.db.publish)({
+      type: 'contact',
+      contact: bob.id,
+      following: true,
+    }),
+
+    pify(alice.db.publish)({
+      type: 'contact',
+      contact: dolores.id,
+      following: true,
+    }),
+    pify(dolores.db.publish)({
+      type: 'contact',
+      contact: alice.id,
+      following: true,
+    }),
+  ])
+  t.pass('published all the messages')
+
+  await sleep(REPLICATION_TIMEOUT)
+
+  await pify(dolores.connect)(alice.getAddress())
+  await pify(dolores.connect)(bob.getAddress())
+
+  await sleep(REPLICATION_TIMEOUT)
+
+  await pify(alice.close)(true)
+  t.pass('stop alice')
+
+  const restartedAlice = createSsbServer({
+    path: path.join(os.tmpdir(), 'server-alice'),
+    keys: aliceKey,
+    timeout: CONNECTION_TIMEOUT,
+    replicationScheduler: {
+      autostart: true,
+      debouncePeriod: 0,
+    },
+    ebt: {
+      logging: true
+    }
+  })
+
+  await pify(restartedAlice.db.publish)({
+    type: 'contact',
+    contact: bob.id,
+    blocking: true,
+  })
+  t.pass('published block')
+
+  await sleep(REPLICATION_TIMEOUT)
+
+  await pify(restartedAlice.db.deleteFeed)(bob.id)
+  await pify(restartedAlice.db.compact)()
+  t.pass('alice removed bob')
+
+  const originalEBTRequest = restartedAlice.ebt.request
+  restartedAlice.ebt.request = function request(id, val) {
+    if (id === bob.id) {
+      if (val === true) {
+        t.fail('requested bob, who is blocked')
+      }
+    }
+    t.pass("alice requested:" + id + ", with:" + val)
+    originalEBTRequest.call(alice.ebt, id, val)
```

Comment:
```suggestion
    originalEBTRequest.call(restartedAlice.ebt, id, val)
```

Not sure if it makes a difference

Reply:
Good catch, doesn't change the result though.